projects/observability/src/pages/apis/service-detail/instrumentation/service-instrumentation.component.ts

```diff
@@ -1,17 +1,28 @@
 import { ChangeDetectionStrategy, Component } from '@angular/core';
 
 import { BreadcrumbsService } from '@hypertrace/components';
-import { Observable } from 'rxjs';
+import { BehaviorSubject, Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
+import { ServiceInstrumentationService } from './service-instrumentation.service';
+import { OrgScoreResponse, ServiceScoreResponse } from './service-instrumentation.types';
 
 @Component({
   styleUrls: ['./service-instrumentation.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [ServiceInstrumentationService],
   template: `
-    <main class="service-instrumentation">
+    <main
+      class="service-instrumentation"
+      [style.visibility]="(this.serviceScoreSubject | async) ? 'visible' : 'hidden'"
```

Comment:
I think we can have a `*ngIf` to control the view logic here and show a `Loader` till subject is not resolved. Also, any reasons behind going with `visibility` and not `display`?

Reply:
Using `display` will disable the animation. The component needs to be initialised with 0 percent and then changed to the actual value for the transition animation to trigger.

If we use display, then the component gets added to the dom with the actual percent value directly and there's no animation. Same problem with ngIf.

Reply:
I will suggest checking [this](https://angular.io/guide/transition-and-triggers) out once as Angular does have a state/transition-type support for blocks. IMO, this should get it done, preferring this over `visibility`, cause `visibility` unnecessarily renders the `DOM` block, which isn't needed.

Reply:
Couldn't find anything in the above link that serves our use case. The animation docs mention applying the animation to a whole block. The element that has the animated property (stroke-dasharray) is nested two levels deep.

![Screenshot 2022-09-13 at 7 53 44 PM](https://user-images.githubusercontent.com/29686866/189927834-7eacbca9-f91b-45cc-9ded-d865d558cb46.png)

Besides, we can't hardcode the CSS values of different states like the docs have shown. The length of the dasharray and color both are calculated based on the response of the API.

Reply:
Tried an alternate approach with `ngIf`. Not sure I'm doing this right but this works. So what we want is to render the component's `circle` initially with a 0 dash length. If we map it to the Input then `ngIf` makes sure the component renders directly at the final state and skips the animation.

```ts
  public internalPercent: BehaviorSubject<number> = new BehaviorSubject<number>(0);
  public dashLengthSubject: BehaviorSubject<string> = new BehaviorSubject<string>('0 999');

  public ngOnInit(): void {
    setTimeout(() => {
      this.internalPercent.next(this.percent);
      this.getDashLength();
    }, 0);
  }
```

I pulled the `getDashLength()` method out of the template and put it inside a `setTimeout` so it doesn't trigger immediately when the component renders, but sometime after. Should I push this code if it looks good?

Reply:
TIL - Will also keep this in mind, thanks.